router.go

```diff
@@ -52,16 +52,19 @@ type MapRouter struct {
 	serviceProcedures         map[serviceProcedure]transport.Procedure
 	serviceProcedureEncodings map[serviceProcedureEncoding]transport.Procedure
 	supportedEncodings        map[serviceProcedure][]string
+	allServices               map[string]bool
 }
 
 // NewMapRouter builds a new MapRouter that uses the given name as the
 // default service name.
-func NewMapRouter(defaultService string) MapRouter {
+func NewMapRouter(defaultService string) (m MapRouter) {
+	defer func() { m.allServices[defaultService] = true }()
```

Comment:
The `defer` seems a bit unnecessary here; (if I remember correctly, `defer`s actually take a small performance hit too)

Two options:
- Create the `MapRouter` first, then update `allServices`, then return; or
- Create a map before the return and assign it to `allServices` in the return

Reply:
Update using the second option.